server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -619,6 +622,34 @@ export function configureWebSocketServices(
 				);
 			}
 
+			// Send signal to room from broadcast-signal endpoint
+			if (httpServer !== undefined && eventEmitter !== undefined) {
+				eventEmitter.on("broadcast-signal", (_, containerUrl) => {
+					const documentId = containerUrl.split("/")[containerUrl.split("/").length - 1];
+					const tenantId = containerUrl.split("/")[containerUrl.split("/").length - 2];
+					const roomFromBroadcastSignal: IRoom = { tenantId, documentId };
+
+					const signalMessageRuntimeMessage: ISignalMessage = {
+						clientId: null, // system signal
+						content: JSON.stringify({
+							type: "RuntimeMessage",
+							contents: {
+								content: {
+									type: "ExternalDataChanged",
+									content: "Data has changed upstream. Please import new data.",
+								},
+								type: "RuntimeMessage",
+							},
```

Comment:
I kind of remember a discussion where we saw that the this whole structure made sense. `type` and `contents` at the top, with `contents.type` and `contents.content.type` and `contents.content.content` all being necessary. Just double checking, in case we can make any of this bits more descriptive / less repetitive.

Reply:
Okay dug through the code again. Here's an image that summarizes it. Looks like the innermost body (lines 638 and 639) isn't necessary for Fluid. So I can get rid of that. 

![Screenshot 2023-08-14 at 10 52 39 AM](https://github.com/microsoft/FluidFramework/assets/6777404/6e8ee5a9-3273-441f-a30e-67496e8c77a9)

1. line 633/634 (toplevel) is required to route to protocolHandler or containerRuntime in the
[loader](https://github.com/microsoft/FluidFramework/blob/main/packages/loader/container-loader/src/container.ts#L2326-L2334) using the logic of clientId === null in
[protocol_handler](https://github.com/microsoft/FluidFramework/blob/main/packages/loader/container-loader/src/protocol.ts#L107-L118)
2. Next, in the [container-runtime](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/container-runtime/src/containerRuntime.ts#L2218-L2222) , we use the top level "content" to cast to the type
[ISignalEnvelope](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/runtime-definitions/src/protocol.ts#L23C2-L41) which is pertinent because it requires a nested:
```
contents: {
    type: string;
    content: any;
  };
```
3. Then last bit necessary for IInboundSignalMessage in [runtime](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/container-runtime/src/containerRuntime.ts#L2218-L2222):
```
    const envelope = message.content as ISignalEnvelope;
    const transformed: IInboundSignalMessage = {
      clientId: message.clientId,
      content: envelope.contents.content,
      type: envelope.contents.type,
    };
```

So we can get rid of the most innermost body (line 638 and 639) here.